### PR TITLE
(feat) Conditionally make the order reason field of the lab order form required when configured

### PR DESCRIPTION
**Pull request by makombe**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR introduces the ability specify if order reason is required for given test(s) which have the order reason configured.

## Screenshots
Screenshot from 2024-02-27 15-48-39

### CODE REVIEW

File: packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx [hunk 1]
```diff
@@ -69,6 +53,34 @@ export function LabOrderForm({
...
+      ? z
+          .string({
+            required_error: translateFrom(moduleName, 'addLabOrderLabOrderReasonRequired', 'Order reason is required'),
+          })
+          .refine(
+            (value) => !!value,
+            translateFrom(moduleName, 'addLabOrderLabOrderReasonRequired', 'Order reason is required'),
+          )
+      : z.string().optional(),
+  });
```
Comment (donaldkibet) [inline packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx:83]: Just a suggestion, could we extract this to a hook e.g `useLabOrderFormSchema`.

**Review by donaldkibet — APPROVED** [review]
LGTM Thanks @makombe, consider incorporating a unit test to prevent regressions

File: packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx (hunk 1 above)
Comment (denniskigen) [reply packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx:83]: You could also separate the constituent parts into individual sub-schemas for readabilty:

```tsx
  const testTypeSchema = z.object(
    {
      label: z.string(),
      conceptUuid: z.string(),
    },
    {
      required_error: translateFrom(moduleName, 'addLabOrderLabTestTypeRequired', 'Test type is required'),
      invalid_type_error: translateFrom(moduleName, 'addLabOrderLabReferenceRequired', 'Test type is required'),
    },
  );

  const orderReasonsSchema = orderReasonRequired
    ? z
        .string({
          required_error: translateFrom(moduleName, 'addLabOrderLabOrderReasonRequired', 'Order reason is required'),
        })
        .refine(
          (value) => !!value,
          translateFrom(moduleName, 'addLabOrderLabOrderReasonRequired', 'Order reason is required'),
        )
    : z.string().optional();

  const labOrderFormSchema = z.object({
    instructions: z.string().optional(),
    urgency: z.string().refine((value) => value !== '', {
      message: translateFrom(moduleName, 'addLabOrderPriorityRequired', 'Priority is required'),
    }),
    labReferenceNumber: z.string().optional(),
    testType: testTypeSchema,
    orderReason: orderReasonsSchema,
  });
```

**Comment by denniskigen** [comment]
Thanks, @makombe!

**Pull request merged**